packages/core/src/connectors/injected.ts

```diff
@@ -93,52 +95,59 @@ export class InjectedConnector extends Connector {
   }
 
   async ready(): Promise<boolean> {
-    /* this.ensureWallet();
+    this.ensureWallet();
 
     if (!this._wallet) return false;
-    return await this._wallet.isPreauthorized();
-    */
 
-    throw new Error("TODO");
+    const permissions: Permission[] = await this.request({
+      type: "wallet_getPermissions",
+    });
+
+    if (!permissions?.includes(Permission.Accounts)) {
+      return false;
+    } else return true;
   }
 
-  async connect(): Promise<ConnectorData> {
+  async account(
+    provider: ProviderOptions | ProviderInterface
+  ): Promise<AccountInterface> {
     this.ensureWallet();
 
     if (!this._wallet) {
-      throw new ConnectorNotFoundError();
+      throw new ConnectorNotConnectedError();
     }
 
-    try {
-      console.log(this._wallet.version);
-      const response = await this._wallet.request({
-        type: "wallet_getPermissions",
-      });
+    const account = new WalletAccount(provider, this._wallet);
+    return account;
+  }
 
-      console.log(response);
-    } catch (err) {
-      console.error(err);
+  async connect(): Promise<ConnectorData> {
+    this.ensureWallet();
+
+    if (!this._wallet) {
+      throw new ConnectorNotFoundError();
     }
 
-    let accounts;
+    let accounts: string[];
     try {
-      accounts = await this._wallet.request({ type: "wallet_requestAccounts" });
+      accounts = await this.request({
+        type: "wallet_requestAccounts",
+        params: { silent_mode: true },
```

Comment:
```suggestion
        params: { silent_mode: false },
```

Not sure about this. I think `silent_mode` means that the user won't be asked to connect the wallet and instead it will just fail, which is not what we want for `connect`. Am I missing something?

Reply:
no i think you are correct, maybe i misunderstood it's purpose. my bad